DEVELOPER_GUIDE.md

```diff
@@ -422,6 +423,88 @@ Output plugins are defined in [flb_output.h](https://github.com/fluent/fluent-bi
 
 The [stdout plugin](plugins/out_stdout) is very simple; review its code to understand how output plugins work.
 
+#### Config Maps
+
+Config maps are an improvement to the previous Fluent Bit API that was used by plugins to read configuration values. The new config maps feature warns the user if there is an unknown configuration key and reduces risk of bad configuration due to typos or deprecated property names. They will also allow dynamic configuration reloading to be implemented in the future.
+
+There are various types of supported configuration types. Full list available [here](https://github.com/fluent/fluent-bit/blob/master/include/fluent-bit/flb_config_map.h#L29). The most used ones are:
```

Comment:
As the referenced line numbers/files in master branch may change in future, to keep the links valid, please reference them by version/commit. For instance, you can replace the link with:

https://github.com/fluent/fluent-bit/blob/v1.4.2/include/fluent-bit/flb_config_map.h#L29